pilot/pkg/model/push_context.go

```diff
@@ -1641,8 +1642,10 @@ func (ps *PushContext) SetDestinationRules(configs []config.Config) {
 
 		rule.Host = string(ResolveShortnameToFQDN(rule.Host, configs[i].Meta))
 		exportToMap := make(map[visibility.Instance]bool)
-		for _, e := range rule.ExportTo {
-			exportToMap[visibility.Instance(e)] = true
+		if rule.GetWorkloadSelector() == nil {
+			for _, e := range rule.ExportTo {
+				exportToMap[visibility.Instance(e)] = true
```

Comment:
this makes sense but can we comment why we do this? its not super obvious

Also can we add validation that if workloadSelector is set exportTo must be `.` or unset? in validation.go

Reply:
Done

Reply:
Do we need to specially check workload selector then?

Reply:
I have added checks in validator.go to make sure that wrong exportTos won't get propagated. This was added as a double check.